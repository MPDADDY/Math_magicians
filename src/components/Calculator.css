.calculator_container {
  border: 1px solid black;
  border-radius: 5px;
  background-color: rgb(9, 60, 105);
  width: 600px;
  height: auto;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  overflow-y: hidden;
}

.buttons {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  height: 100px;
}

.operators {
  background-color: rgb(255, 166, 71);
}

::placeholder {
  /* Chrome, Firefox, Opera, Safari 10.1+ */
  color: white;
  opacity: 1;

  /* Firefox */
}

.inputText {
  background-color: gray;
  width: 100%;
  height: 100px;
  outline: none;
  text-align: right;
  font-size: 30px;
  font-weight: 400;
  padding: 20px;
  color: white;
  cursor: none;
  pointer-events: none;
}

input[type=button] {
  font-size: 20px;
  font-weight: 800;
}

input[type=button]:hover {
  cursor: pointer;
}

.buttonsLast {
  display: grid;
  grid-template-columns: 2fr repeat(2, 1fr);
  height: 100px;
}

.display {
  width: 100%;
  height: 80px;
  background-color: gray;
}

.calcButtons {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  height: 100%;
}
