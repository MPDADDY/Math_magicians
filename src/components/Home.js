function Home() {
  return (
    <div className="home">
      <h1>Welcome on board</h1>
      <p>This is a React web App  for arithmetic and scientific calculations.</p>
      <p>Navigate to calculator whenever you want to do some atithmetic calculations</p>
    </div>
  );
}
export default Home;
