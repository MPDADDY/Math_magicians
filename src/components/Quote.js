import React, { useState, useEffect } from 'react';

function Quote() {
  const [quoteD, setQuoteD] = useState({
    quote: '',
    author: '',
    loading: true,
    error: '',
  });

  useEffect(() => {
    const category = 'fear';
    const apiKey = 'QhIZ4yn9aAJns00FmcPjkQ==7UzA0vhMAkRAAu0i';

    // Construct the API URL with the category
    const apiUrl = `https://api.api-ninjas.com/v1/quotes?category=${category}`;

    fetch(apiUrl, {
      method: 'GET',
      headers: {
        'X-Api-Key': apiKey,
      },
    })
      .then((res) => {
        if (!res.ok) {
          throw new Error('Failed to fetch data');
        }
        return res.json();
      })
      .then((data) => {
        if (data.length > 0) {
          setQuoteD({
            quote: data[0].quote,
            author: data[0].author,
            loading: false,
            error: null,
          });
        } else {
          throw new Error('No data found');
        }
      })
      .catch((err) => {
        setQuoteD({
          quote: null,
          author: null,
          loading: false,
          error: err.message,
        });
      });
  }, []);

  const {
    loading, error, quote, author,
  } = quoteD;

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        Error:
        {error}
      </div>
    );
  }

  return (
    <div>
      <h2>Today&apos;s quote</h2>
      <p>{quote}</p>
      <p>{author}</p>
    </div>
  );
}

export default Quote;
