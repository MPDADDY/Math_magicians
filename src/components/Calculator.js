import React from 'react';
import './Calculator.css';
import InputText from './InputText';

const Calculator = () => (
  <section className="calculator_container">
    <form action="">
      <div>
        <InputText />
      </div>

      <article className="calcButtons_container">
        <div className="buttons">
          <input type="button" value="AC" />
          <input type="button" value="+/-" />
          <input type="button" value="%" />
          <input className="operators" type="button" value="&divide;" />
        </div>
        <div className="buttons">
          <input type="button" value="7" />
          <input type="button" value="8" />
          <input type="button" value="9" />
          <input className="operators" type="button" value="&times;" />
        </div>
        <div className="buttons">
          <input type="button" value="4" />
          <input type="button" value="5" />
          <input type="button" value="6" />
          <input className="operators" type="button" value="&minus;" />
        </div>
        <div className="buttons">
          <input type="button" value="1" />
          <input type="button" value="2" />
          <input type="button" value="3" />
          <input className="operators" type="button" value="&#43;" />
        </div>
        <div className="buttonsLast">
          <input type="button" value="0" />
          <input type="button" value="&#46;" />
          <input className="operators" type="button" value="=" />
        </div>
      </article>
    </form>
  </section>
);

export default Calculator;
