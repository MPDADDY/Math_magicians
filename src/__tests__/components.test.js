import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Quote from '../components/Quote';

describe('components renders correctly', () => {

  it('renders the Home component with content', () => {
    const result = Home();
    expect(result).toEqual(
      <div className="home">
        <h1>Welcome on board</h1>
        <p>This is a React web App  for arithmetic and scientific calculations.</p>
        <p>Navigate to calculator whenever you want to do some atithmetic calculations</p>
      </div>,
    );
  });

  it('home components renders correctly', () => {
    const tree = renderer
    .create(<Home page="/">Home</Home>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calculator components renders correctly', () => {
    const tree = renderer
    .create(<Calculator page="/Calculator">Calculator</Calculator>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('quote components renders correctly', () => {
    const tree = renderer
    .create(<Quote page="/Quote">Quotes</Quote>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
