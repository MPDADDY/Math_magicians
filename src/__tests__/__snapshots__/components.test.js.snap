// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components renders correctly calculator components renders correctly 1`] = `
<section
  className="calculator_container"
>
  <form
    action=""
  >
    <div>
      <input
        className="inputText"
        onClick={[Function]}
        type="button"
        value="0"
      />
    </div>
    <article
      className="calcButtons_container"
    >
      <div
        className="buttons"
      >
        <input
          onClick={[Function]}
          type="button"
          value="AC"
        />
        <input
          onClick={[Function]}
          type="button"
          value="+/-"
        />
        <input
          onClick={[Function]}
          type="button"
          value="%"
        />
        <input
          className="operators"
          onClick={[Function]}
          type="button"
          value="÷"
        />
      </div>
      <div
        className="buttons"
      >
        <input
          onClick={[Function]}
          type="button"
          value="7"
        />
        <input
          onClick={[Function]}
          type="button"
          value="8"
        />
        <input
          onClick={[Function]}
          type="button"
          value="9"
        />
        <input
          className="operators"
          onClick={[Function]}
          type="button"
          value="×"
        />
      </div>
      <div
        className="buttons"
      >
        <input
          onClick={[Function]}
          type="button"
          value="4"
        />
        <input
          onClick={[Function]}
          type="button"
          value="5"
        />
        <input
          onClick={[Function]}
          type="button"
          value="6"
        />
        <input
          className="operators"
          onClick={[Function]}
          type="button"
          value="−"
        />
      </div>
      <div
        className="buttons"
      >
        <input
          onClick={[Function]}
          type="button"
          value="1"
        />
        <input
          onClick={[Function]}
          type="button"
          value="2"
        />
        <input
          onClick={[Function]}
          type="button"
          value="3"
        />
        <input
          className="operators"
          onClick={[Function]}
          type="button"
          value="+"
        />
      </div>
      <div
        className="buttonsLast"
      >
        <input
          onClick={[Function]}
          type="button"
          value="0"
        />
        <input
          onClick={[Function]}
          type="button"
          value="."
        />
        <input
          className="operators"
          onClick={[Function]}
          type="button"
          value="="
        />
      </div>
    </article>
  </form>
</section>
`;

exports[`components renders correctly home components renders correctly 1`] = `
<div
  className="home"
>
  <h1>
    Welcome on board
  </h1>
  <p>
    This is a React web App  for arithmetic and scientific calculations.
  </p>
  <p>
    Navigate to calculator whenever you want to do some atithmetic calculations
  </p>
</div>
`;

exports[`components renders correctly quote components renders correctly 1`] = `
<div>
  Loading...
</div>
`;
